packages/bpk-component-horizontal-nav/src/bpk-horizontal-nav-item.scss

```diff
@@ -40,21 +40,25 @@
     cursor: pointer;
 
     @include bpk-themeable-property(color, --bpk-horizontal-nav-link-color, $bpk-horizontal-nav-link-color);
+    @include bpk-themeable-property(fill, --bpk-horizontal-nav-link-color, $bpk-horizontal-nav-link-color);
 
     @include bpk-hover {
       @include bpk-border-bottom-xl($bpk-color-gray-100);
       @include bpk-themeable-property(color, --bpk-horizontal-nav-link-hover-color, $bpk-horizontal-nav-link-hover-color);
+      @include bpk-themeable-property(fill, --bpk-horizontal-nav-link-hover-color, $bpk-horizontal-nav-link-hover-color);
     }
 
     &:active {
       @include bpk-border-bottom-xl($bpk-color-gray-200);
       @include bpk-themeable-property(color, --bpk-horizontal-nav-link-active-color, $bpk-horizontal-nav-link-active-color);
+      @include bpk-themeable-property(fill, --bpk-horizontal-nav-link-active-color, $bpk-horizontal-nav-link-active-color);
     }
 
     &--selected {
       @include bpk-border-bottom-xl($bpk-horizontal-nav-bar-selected-color);
       @include bpk-border-bottom-xl(var(--bpk-horizontal-nav-bar-selected-color, $bpk-horizontal-nav-bar-selected-color));
       @include bpk-themeable-property(color, --bpk-horizontal-nav-link-selected-color, $bpk-horizontal-nav-link-selected-color);
+      @include bpk-themeable-property(fill, --bpk-horizontal-nav-link-selected-color, $bpk-horizontal-nav-link-selected-color);
```

Comment:
You should be able to achieve the same thing with `fill: currentColor`, will remove a fair bit of code. Can you give it a go and see if it works?